### DISCUSSION TITLE
Just need to lazy load images. No optimization required. Can I use `next/image`?

### DISCUSSION THREAD

**Discussion by msafi** [vercel/next.js#37186] [Help] [answered]
I want to use `next/image` to display the Open Graph image of 3rd party websites. The URLs for these images is only known at runtime, not before that, so the domains cannot be pre-configured.

I don't need Next.js to serve these images. They're already hosted by 3rd parties and expected to be retrieved from anywhere. I just want to use `next/image` for features such as lazy loading.

Is that possible?

**Comment by icyJoseph** [comment, marked as answer]
I guess if you use a custom loader, which just returns the same URL it gets passed on, and you set the [`unoptimized`](https://nextjs.org/docs/api-reference/next/image#unoptimized) flag to true, then you'd get this behaviour.

Next.js would fetch the image for you and just pass it along.

A loader looks like this:

```ts

const customLoader: ImageLoader = ({ src, width, quality }) => {
  const url = new URL(src);
  const pathname = url.pathname;
  const [, id] = pathname.split("/").filter(Boolean);

  return `${url.origin}/id/${id}/${width}`;
};
```

Give it a go, I don't have time right now to set up a test bench for this thing in particular.

**Reply by msafi** [reply]
That worked. Thank you!

This is the code I used

```tsx
<Image
  loader={({ src }) => src}
  src="https://image.cnbcfm.com/api/v1/image/106926992-1628885077267-elon.jpg?v=1644936558&w=1920&h=1080"
/>
```
